src/utils/IndexKeyUtils.cpp

```diff
@@ -144,7 +128,25 @@ IndexKeyUtils::collectIndexValues(RowReader* reader,
                        << ". status : " << ret;
             return ret;
         }
-        values.emplace_back(std::move(v));
+        if (isNullable && v.isNull()) {
+            values.emplace_back(Value(NullType::__NULL__));
+            continue;
+        } else if (!isNullable && v.isNull()) {
+            LOG(ERROR) << "Field cannot be NULL by : " << col.get_name();
+            return Status::NotSupported("Field cannot be NULL by : %s", col.get_name().c_str());
+        }
+        if (col.type.get_type() == meta::cpp2::PropertyType::FIXED_STRING) {
+            std::string fs = v.getStr();
+            auto len = static_cast<size_t>(*col.type.get_type_length());
+            if (len > v.getStr().size()) {
+                fs.append(len - v.getStr().size(), '\0');
+            } else {
+                fs = fs.substr(0, len);
+            }
+            values.emplace_back(Value(fs));
+        } else {
+            values.emplace_back(std::move(v));
+        }
```

Comment:
A small question:
If the result of `checkValue `  is ok,   just judge whether the result of v.isNull() is null. WDYT?

Reply:
> A small question:
> If the result of `checkValue ` is ok, just judge whether the result of`v.isNull()` is true or false. WDYT?

Good point. you are right.

Reply:
improved.